### ISSUE TITLE
Make MeshStreamer classes [migratable] to support checkpoint/restart

### ISSUE THREAD

**Issue by PhilMiller** [UIUC-PPL/charm#647]
*Original issue: https://charm.cs.illinois.edu/redmine/issues/647*

---
Right now, the various chare classes defined in `NDMeshStreamer.h` lack the `[migratable]` attribute. This means that programs using TRAM will fail to checkpoint. They should have the attribute and matching pup methods added.

**Comment by harshithamenon** [comment]
*Original date: 2015-10-06 20:45:41*

---
There are couple of examples for using TRAM in examples/charm++/TRAM. This does not contain checkpoint restart though.

**Comment by karthiksenthil** [comment]
*Original date: 2017-04-27 00:29:14*

---
I have completed adding PUP functions for all the involved classes in `VirtualRouter.h` and for the `MeshStreamer` base class and `GroupMeshStreamer` class in NDMeshStreamer.h

I'm a bit unsure about the technique to handle PUP for the data member : `std::vector<std::vector<MeshStreamerMessage<dtype> * > > dataBuffers_`. Here MeshStreamerMessage is a Charm++ message object templated on a datatype(dtype).

I've implemented the unpacking operation for this as follows:


```
<code class="cpp">
if (p.isUnpacking()) {
      dataBuffers_.resize(numDimensions_);
      for (int i = 0; i < numDimensions_; i++) {
          dataBuffers_[i].assign(myRouter_.numBuffersPerDimension(i), (MeshStreamerMessage<dtype> *) NULL);
    }
}
</code>
```



This is similar to the operation done in the constructor(MeshStreamer::ctorHelper). Is this correct? I don't get any compiler or runtime errors while testing available TRAM examples/tests on my machine. Also, what should be the packing operation on this data member?
